onnxruntime/core/providers/cpu/ml/scaler.cc

```diff
@@ -85,7 +87,7 @@ common::Status ScalerOp<T>::Compute(OpKernelContext* context) const {
   int64_t stride = x_dims.size() == 1 ? x_dims[0] : x_dims[1];
   auto* ttp = context->GetOperatorThreadPool();
   auto conditional_batch_call = [ttp, x_size](std::function<void(ptrdiff_t)> f) {
-    if (x_size < 10 * 1000) {  // TODO: tune this, arbitrary threshold
+    if (x_size < kParallelizationThreshold) {  // TODO: tune this, arbitrary threshold
```

Comment:
I wonder if this decision could not be handled in ThreadPool. TryBatchParallelFor can handle a more complex cost.

Reply:
Yes, once we learn more we can convert this into a proper cost based model. For now, just relying on a crude heuristic.